src/app/pages/EditPage/EditPage.tsx

```diff
@@ -0,0 +1,156 @@
+import useFetch from '../../hooks/useFetch';
+import type { Gear } from '../../../types';
+
+export default function EditPage(): JSX.Element {
+  const history = useHistory();
+  const params = useParams<{ name: string }>();
+
+  const { data: gear, isLoading } = useFetch<Gear>(`/api/gear/${params.name}`);
+
+  const [selectedGear, setSelectedGear] = useState<'Drum' | 'Synth' | 'Effect'>(
+    'Synth'
+  );
+
+  const [output, setOutput] = useState<string>('');
+  const [isOutputChecked, setIsOutputChecked] = useState(false);
+
+  const [input, setInput] = useState<string>('');
+  const [isInputChecked, setIsInputChecked] = useState(false);
+
+  const [midi_in, setMidi_in] = useState<string>('');
+  const [isMidi_inChecked, setIsMidi_inChecked] = useState(false);
+
+  const [midi_out, setMidi_out] = useState<string>('');
+  const [isMidi_outChecked, setIsMidi_outChecked] = useState(false);
+
+  const [usb, setUsb] = useState<string>('');
+  const [isUsbChecked, setIsUsbChecked] = useState(
+    gear && gear.connections[4] ? true : false
+  );
+
+  useEffect(() => {
+    if (gear) {
+      setSelectedGear(gear.iconType);
+      setIsOutputChecked(gear.connections[0] ? true : false);
+      setIsInputChecked(gear.connections[1] ? true : false);
+      setIsMidi_inChecked(gear.connections[2] ? true : false);
+      setIsMidi_outChecked(gear.connections[3] ? true : false);
+      setIsUsbChecked(gear.connections[4] ? true : false);
+    }
+  }, [gear]);
+
+  async function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+
+    const connections = [
+      isOutputChecked ? { channel: output, connection: 'Output to' } : '',
+      isInputChecked ? { channel: input, connection: 'Input from' } : '',
+      isMidi_inChecked ? { channel: midi_in, connection: 'MIDI In from' } : '',
+      isMidi_outChecked
+        ? { channel: midi_out, connection: 'MIDI Out from' }
+        : '',
+      isUsbChecked ? { channel: usb, connection: 'USB Port' } : '',
+    ];
+
+    const editGear = {
+      iconType: selectedGear,
+      connections: connections.filter((connection) => connection),
+    };
+
+    const response = await fetch(`/api/gear/${params.name}`, {
+      method: 'PATCH',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(editGear),
+    });
+    await response.json();
+    history.push('/');
+  }
+
+  function goBack() {
+    history.push('/');
+  }
+
+  return (
+    <div className={style.pageContainer}>
+      <Header withBackButton text="Edit" onClick={goBack} />
+
+      <main className={style.main}>
+        {!isLoading && gear && (
+          <>
+            <GearSelector selected={selectedGear} onChange={setSelectedGear} />
+            <form onSubmit={(event) => handleSubmit(event)}>
+              <GearInput
+                type="text"
+                placeholder={
+                  gear && gear ? gear?.connections[0].channel : 'Output to'
```

Comment:
I'm sorry, but I think usability wise it would be better to the sates `output`, `input` etc. instead of this